src/sqlfluff/core/parser/segments/base.py

```diff
@@ -131,6 +131,9 @@ def __eq__(self, other):
             )
         )
 
+    def __hash__(self):
+        return hash((self.__class__.__name__, self.raw, self.pos_marker.source_position()))
+
```

Comment:
Add `__hash__` for `set.add` function which need hashable for remove duplicate element.